src/command_modules/azure-cli-eventgrid/azure/cli/command_modules/eventgrid/custom.py

```diff
@@ -50,15 +68,76 @@ def cli_topic_create_or_update(
         topic_name,
         location,
         tags=None):
+    topic_info = Topic(
+        location=location,
+        tags=tags)
     async_topic_create = client.create_or_update(
         resource_group_name,
         topic_name,
+        topic_info)
+    created_topic = async_topic_create.result()
+    return created_topic
+
+
+def cli_domain_list(
+        client,
+        resource_group_name=None,
+        odata_query=None):
+
+    if resource_group_name:
+        return client.list_by_resource_group(resource_group_name, odata_query, DEFAULT_TOP)
+
+    return client.list_by_subscription(odata_query, DEFAULT_TOP)
+
+
+def cli_domain_create_or_update(
+        client,
+        resource_group_name,
+        domain_name,
         location,
-        tags)
+        tags=None):
+    domain_info = Domain(
+        location=location,
+        tags=tags)
 
-    created_topic = async_topic_create.result()
+    async_domain_create = client.create_or_update(
+        resource_group_name,
+        domain_name,
+        domain_info)
+    created_domain = async_domain_create.result()
+    return created_domain
```

Comment:
Don't call result yourself. The CLI handles this. You simply return the value of the create_or_update call (which would be a poller).

Reply:
Fixed here and for existing code.